app/models/task.rb

```diff
@@ -18,40 +11,38 @@ class Task < ActiveRecord::Base
     started: :finished,
     danger: :finished
   }.stringify_keys.freeze
-  Updaters = %w[github pivotal local preliminary development post].freeze
+  Updaters = %w[github pivotal local requesting planning execution delivering].freeze
   UPDATER = {
     github: GithubUpdater,
     pivotal: PivotalUpdater,
     local: LocalUpdater,
-    preliminary: PreliminaryUpdater,
-    development: DevelopmentUpdater,
-    post: PostUpdater
+    requesting: RequestingUpdater,
+    planning: PlanningUpdater,
+    execution: ExecutionUpdater,
+    delivering: DeliveringUpdater,
+    testing: TestingUpdater
   }.stringify_keys.freeze
-  DevTaskTitles = ['Lo-fi Mockup', 'Pair programming', 'Code Review',
-                   'Finish Story', 'TDD and BDD', 'Points Estimation',
-                   'Pull Request'].freeze
-  PreTaskTitles = ['Customer Meeting', 'Iteration Planning', 'GSI Meeting',
-                   'Scrum meeting', 'Configuration Setup', 'Test Title'].freeze
-  PostTaskTitles = ['Deploy', 'Cross Group Review', 'Customer Feedback'].freeze
+  Requesting = ['Contact Customer', 'Customer Meeting', 'Create Stories'].freeze
+  Planning = ['Planning Meetings', 'Behavior Tests'].freeze
+  Execution = ['Unit Tests', 'Implementation'].freeze
+  Delivering = ['Pull Requests', 'Code Review'].freeze
   validates :task_status, presence: true, inclusion: { in: Status }
-  validates :updater_type, presence: true, inclusion: { in: Updaters }
-  validates :title, inclusion: { in: PreTaskTitles | DevTaskTitles | PostTaskTitles }
+  validates :updater_type, presence: true, inclusion: { in: Updaters | ["testing"]}
+  validates :title, inclusion: { in: Requesting | Planning | Execution | Delivering | ["test task"]}
 
-  def self.abstract_graph(start_task)
-    children = Taskedge.find_children(start_task)
-    graph = Hash.new
-    visited = Array.new(children)
-    visited.append(start_task)
-    graph[start_task] = Array.new(children)
-    until children.empty?
-      newnode = children.shift
-      newchildren = Taskedge.find_children newnode
-      graph[newnode] = Array.new(newchildren)
-      newchildren.delete_if{|child| visited.include? child}
-      visited.concat(newchildren)
-      children.concat(newchildren)
-    end
-    JSON.generate(graph)
+  # Static constant defined by the type of updater
+  def self.phases_task
+    return Requesting, Planning, Execution, Delivering
+  end
+
+  # According to iteration separate all tasks according to their updater type(categories)
+  def self.tasks_selection iteration
+    tasks = Task.where(iteration: iteration.id)
```

Comment:
How many db queries are made here?  I suspect that you're making four separate queries, as you're not calling `all` here.

In general, we'd like to get all the data in one query.  DB queries can easily become a bottleneck of response time.

Reply:
I think the tasks here is already list,not sure